src/test/scala/advancepassengerinfo/importer/processor/DqFileProcessorImplTest.scala

```diff
@@ -216,46 +218,37 @@ class DqFileProcessorTest extends TestKit(ActorSystem("MySpec"))
-          Future.successful(List(fallbackFileName, "3.zip")),
-        )).map(_.flatten)
+    "handle exceptions while getting nextFiles and start again" in {
+      def createManifests(manifest: VoyageManifest): Seq[(String, Try[VoyageManifest])] = Seq(
+        ("manifest1.json", Success(manifest)),
+        ("manifest2.json", Success(manifest)),
+      )
 
-        val s3Files: String => Future[List[String]] = previous => Future.sequence(List(
+      def createMockFileNamesProvider(count: AtomicInteger): FileNames = new FileNames {
+        private val s3Files: String => Future[List[String]] = previous => Future.sequence(List(
           Future.successful(List("1.zip", "2.zip")),
-          Future.failed(new Exception("next file exception")),
+          if (count.get() == 0) {
+            count.incrementAndGet()
+            Future.failed(new Exception("next file exception"))
+          } else Future.successful(List("3.zip")),
         )).map(_.flatten)
 
-        override val nextFiles: (String, String) => Future[List[String]] = (lastFile: String, fallbackFileName) => s3Files(lastFile)
-          .recoverWith {
-            case _ =>
-              fallBackListFiles(fallbackFileName)
-          }.map(_.filterNot(_ == lastFile))
+        override val nextFiles: (String) => Future[List[String]] = (lastFile: String) => s3Files(lastFile)
       }
+
+      val mockFileNamesProvider = createMockFileNamesProvider(new AtomicInteger(0))
       val probe = TestProbe("probe")
       val mockPersistence = MockPersistence(probe.ref)
-
-      val manifests = Seq(
-        (s"manifest1.json", Success(manifest)),
-        (s"manifest2.json", Success(manifest)),
-      )
+      val manifests = createManifests(manifest)
       val processor = DqFileProcessorImpl(singleZipMockProvider(manifests), mockPersistence)
-      val dqApiFeed: DqApiFeedImpl = DqApiFeedImpl(
-        mockFileNamesProvider,
-        processor,
-        100.millis,
-        MockStatsDCollector
-      )
+      val dqApiFeed: DqApiFeedImpl = DqApiFeedImpl(mockFileNamesProvider, processor, 100.millis, MockStatsDCollector, LastCheckedState())
 
       dqApiFeed.processFilesAfter("1.zip").runWith(Sink.seq)
       probe.expectMsg(ManifestCall("manifest1.json", manifest))
-      probe.expectMsg(JsonFileCall(fallbackFileName, s"manifest1.json", successful = true, dateIsSuspicious = false))
+      probe.expectMsg(JsonFileCall("2.zip", "manifest1.json", successful = true, dateIsSuspicious = false))
       probe.expectMsg(ManifestCall("manifest2.json", manifest))
-      probe.expectMsg(JsonFileCall(fallbackFileName, "manifest2.json", successful = true, dateIsSuspicious = false))
-      probe.expectMsg(ZipFileCall(fallbackFileName, successful = true))
-
+      probe.expectMsg(JsonFileCall("2.zip", "manifest2.json", successful = true, dateIsSuspicious = false))
+      probe.expectMsg(ZipFileCall("2.zip", successful = true))
     }
 
     "handle exception while persisting" in {
```

Comment:
This test also needs changing so we can see it carries on after a failure. It should be testing the other recover you've added